tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -75,7 +117,7 @@ object CodeGen {
       import build.Mode._
       (config.mode, config.LTO) match {
         case (Debug, _)                                  => separate()
-        case (ReleaseFast | ReleaseFull, build.LTO.None) => single()
+        case (ReleaseFast | ReleaseFull, build.LTO.None) => separate()
```

Comment:
release + LTO.none should still be generated to a single file (for performance reasons in non lto builds) This configuration should also be checked at earlier stages so we don't introduce any additional and not utilize overhead. In such a case a warning should also be logged for the users, to make them aware of this limitation.